drivers/modem/gsm_ppp.c

```diff
@@ -1030,6 +1037,26 @@ void gsm_ppp_stop(const struct device *dev)
 				      K_SECONDS(10))) {
 		LOG_WRN("Failed locking modem cmds!");
 	}
+
+	if (gsm->modem_off_cb) {
+		gsm->modem_off_cb(NULL);
+	}
+}
+
+void gsm_ppp_register_modem_on_callback(const struct device *dev,
+					void (*gsm_modem_on_cb)(void *user_data))
```

Comment:
You need to add user_data parameter too here.
Also it would make the prototype more readable if you create a typedef for the modem cb and use that here.
Instead of separate on/off callbacks, we could just do one function that sets both callbacks, then handling of user_data is more clearer. So something like
```
void gsm_ppp_register_modem_power_callback(const struct device *dev,
				gsm_modem_power_cb on, gsm_modem_power_cb off, void *user_data)
```

Reply:
Done.